### ISSUE TITLE
vnstat 2.7 makes weird trafficcount on PPPoE interface when it's reconnecting

### ISSUE THREAD

**Issue by Hornochs** [vergoh/vnstat#224]
Hey there,

on my pfSense 2.5.2 is a weird behaviour on my pppoe0 interface:

Since I'm a monitoring freak I setup a grafana board and I'm finally able to export my traffic stats to an influxdb (asked this question on this board aswell, solution will be posted shortly.)

But now here is the funny part: My ISP of my PPPoE disconnects me forcefully every 24h. Therefore I setup my pfsense to forcefully disconnect at 4am every morning. Here is the strange part: vnStat thinks, that when I'm reconnecting I have a traffic of 4 GB:

![grafik](https://user-images.githubusercontent.com/63955136/153749610-f6379614-f929-497c-923d-1e5de79ffa8d.png)

Does anybody has an idea why? I mean, that shouldn't be normal.

**Comment by vergoh** [comment]
I suspect every time that PPPoE interface reconnects it will also reset the traffic counters for that interface back to zero. It's also possible the interface will not be available for a short duration or that will pretty much happen instantly. In either case, vnStat doesn't see that change happening (interface going missing and then coming back) so it has to use other methods to guess if the seen traffic (due to the counters appearing like they've overflown) is real or if there was some sort of counter reset (like is the case with PPPoE).

Could you list the values for the following options from the configuration file:

- BandwidthDetection
- MaxBandwidth
- MaxBWpppoe0 (or that may also be MaxBWppp0 depending on how the interface is really named)
- UpdateInterval
- PollInterval

\+ the information how fast the actual connection for that PPPoE interface is.

Usually this sort of issues can be solved by lowering `UpdateInterval` and setting the correct MaxBandwidth for the interface (if it isn't automatically detected correctly). The default values depend somewhat on how those have been set in the likely binary distribution you are using so I'm not even trying to guess.

**Comment by Hornochs** [comment]
Is there a way where I can see where the config file is in pfSense?

Because I only find a file in /usr/local/etc/vnstat.conf:
`#########################################
# vnStat 2.2 config file                #
# AUTOMATICALLY GENERATED. DO NOT EDIT! #
#########################################

# Location of the database directory
DatabaseDir "/var/db/vnstat"

# On which day should months change
MonthRotate 1

# Image colors
CBackground     "FFFFFF"
CEdge           "AEAEAE"
CHeader         "606060"
CHeaderTitle    "FFFFFF"
CHeaderDate     "FFFFFF"
CText           "000000"
CLine           "B0B0B0"
CLineL          "-"
CRx             "92CF00"
CTx             "606060"
CRxD            "-"
CTxD            "-"
`

**Comment by Hornochs** [comment]
Ah alright, sure I got shell access. Here is the result:
```
# vnStat 2.7 config file
##

# default interface (leave empty for automatic selection)
Interface ""

# location of the database directory
DatabaseDir "/var/db/vnstat"

# locale (LC_ALL) ("-" = use system locale)
Locale "-"

# date output formats for -d, -m, -t and -w
DayFormat    "%Y-%m-%d"
MonthFormat  "%Y-%m"
TopFormat    "%Y-%m-%d"

# characters used for visuals
RXCharacter       "%"
TXCharacter       ":"
RXHourCharacter   "r"
TXHourCharacter   "t"

# how units are prefixed when traffic is shown
# 0 = IEC standard prefixes (KiB/MiB/GiB...)
# 1 = old style binary prefixes (KB/MB/GB...)
# 2 = SI decimal prefixes (kB/MB/GB...)
UnitMode 0

# used rate unit (0 = bytes, 1 = bits)
RateUnit 1

# how units are prefixed when traffic rate is shown in bits
# 0 = IEC binary prefixes (Kibit/s...)
# 1 = SI decimal prefixes (kbit/s...)
RateUnitMode 1

# output style
# 0 = minimal & narrow, 1 = bar column visible
# 2 = same as 1 except rate in summary
# 3 = rate column visible
OutputStyle 3
EstimateBarVisible 1

# number of decimals to use in outputs
DefaultDecimals 2
HourlyDecimals 1

# spacer for separating hourly sections (0 = none, 1 = '|', 2 = '][', 3 = '[ ]')
HourlySectionStyle 2

# how many seconds should sampling for -tr take by default
Sampletime 5

# default query mode
# 0 = normal, 1 = days, 2 = months, 3 = top, 5 = short
# 7 = hours, 8 = xml, 9 = one line, 10 = json
QueryMode 0

# default list output entry limits (0 = all)
List5Mins      24
ListHours      24
ListDays       30
ListMonths     12
ListYears       0
ListTop        10


# vnstatd
##

# switch to given user when started as root (leave empty to disable)
DaemonUser ""

# switch to given group when started as root (leave empty to disable)
DaemonGroup ""

# try to detect interface maximum bandwidth, 0 = disable feature
# MaxBandwidth will be used as fallback value when enabled
BandwidthDetection 1

# maximum bandwidth (Mbit) for all interfaces, 0 = disable feature
# (unless interface specific limit is given)
MaxBandwidth 1000

# interface specific limits
#  example 8Mbit limit for eth0 (remove # to activate):
#MaxBWeth0 8

# data retention durations (-1 = unlimited, 0 = feature disabled)
5MinuteHours   48
HourlyDays      4
DailyDays      62
MonthlyMonths  25
YearlyYears    -1
TopDayEntries  20

# how often (in seconds) interface data is updated
UpdateInterval 20

# how often (in seconds) interface status changes are checked
PollInterval 5

# how often (in minutes) data is saved to database
SaveInterval 5

# how often (in minutes) data is saved when all interface are offline
OfflineSaveInterval 30

# on which day should months change
MonthRotate 1
MonthRotateAffectsYears 0

# filesystem disk space check (1 = enabled, 0 = disabled)
CheckDiskSpace 1

# how much the boot time can variate between updates (seconds)
BootVariation 15

# create database entries even when there is no traffic (1 = enabled, 0 = disabled)
TrafficlessEntries 1

# how many minutes to wait during daemon startup for system clock to
# sync time if most recent database update appears to be in the future
TimeSyncWait 5

# how often (in minutes) bandwidth detection is done when
# BandwidthDetection is enabled (0 = disabled)
BandwidthDetectionInterval 5

# force data save when interface status changes (1 = enabled, 0 = disabled)
SaveOnStatusChange 1

# enable / disable logging (0 = disabled, 1 = logfile, 2 = syslog)
UseLogging 2

# create dirs if needed (1 = enabled, 0 = disabled)
CreateDirs 1

# update ownership of files if needed (1 = enabled, 0 = disabled)
UpdateFileOwner 1

# file used for logging if UseLogging is set to 1
LogFile "/var/log/vnstat/vnstat.log"

# file used as daemon pid / lock file
PidFile "/var/run/vnstat/vnstat.pid"

# 1 = 64-bit, 0 = 32-bit, -1 = old style logic, -2 = automatic detection
64bitInterfaceCounters -2

# use SQLite Write-Ahead Logging mode (1 = enabled, 0 = disabled)
DatabaseWriteAheadLogging 0

# change the setting of the SQLite "synchronous" flag
# (-1 = auto, 0 = off, 1, = normal, 2 = full, 3 = extra)
DatabaseSynchronous -1


# vnstati
##

# title timestamp format
HeaderFormat "%Y-%m-%d %H:%M"

# show hours with rate (1 = enabled, 0 = disabled)
HourlyRate 1

# show rate in summary (1 = enabled, 0 = disabled)
SummaryRate 1

# transparent background (1 = enabled, 0 = disabled)
TransparentBg 0

# image size control
LargeFonts 0
LineSpacingAdjustment 0
ImageScale 100

# 5 minutes graph size control
5MinuteGraphResultCount 576
5MinuteGraphHeight 300

# horizontal/vertical summary graph (0 = hours, 1 = 5 minutes)
SummaryGraph 0

# traffic estimate bar style
# (0 = not shown, 1 = continuation of existing bar, 2 = separate bar)
EstimateStyle 1

# bar column in list outputs shows rate if OutputStyle is 3
# (1 = enabled, 0 = disabled)
BarColumnShowsRate 0

# image colors
CBackground     "FFFFFF"
CEdge           "AEAEAE"
CHeader         "606060"
CHeaderTitle    "FFFFFF"
CHeaderDate     "FFFFFF"
CText           "000000"
CLine           "B0B0B0"
CLineL          "-"
CRx             "92CF00"
CTx             "606060"
CRxD            "-"
CTxD            "-"
```
and with -D:
```
Debug enabled, vnstat 2.7
Config file: /usr/local/etc/vnstat.conf
  c: DatabaseDir "/var/db/vnstat"
   -> "DatabaseDir": "/var/db/vnstat"
  i: MonthRotate 1
   -> "MonthRotate": 1
  c: CBackground     "FFFFFF"
   -> "CBackground": "FFFFFF"
  c: CEdge           "AEAEAE"
   -> "CEdge": "AEAEAE"
  c: CHeader         "606060"
   -> "CHeader": "606060"
  c: CHeaderTitle    "FFFFFF"
   -> "CHeaderTitle": "FFFFFF"
  c: CHeaderDate     "FFFFFF"
   -> "CHeaderDate": "FFFFFF"
  c: CText           "000000"
   -> "CText": "000000"
  c: CLine           "B0B0B0"
   -> "CLine": "B0B0B0"
  c: CLineL          "-"
   -> "CLineL": "-"
  c: CRx             "92CF00"
   -> "CRx": "92CF00"
  c: CTx             "606060"
   -> "CTx": "606060"
  c: CRxD            "-"
   -> "CRxD": "-"
  c: CTxD            "-"
   -> "CTxD": "-"
Config file: /usr/local/etc/vnstat.conf
ibw list is empty.
arg 1: "-D"
arg 2: "--showconfig"
# vnStat 2.7 config file
##

# default interface (leave empty for automatic selection)
Interface ""

# location of the database directory
DatabaseDir "/var/db/vnstat"

# locale (LC_ALL) ("-" = use system locale)
Locale "-"

# date output formats for -d, -m, -t and -w
DayFormat    "%Y-%m-%d"
MonthFormat  "%Y-%m"
TopFormat    "%Y-%m-%d"

# characters used for visuals
RXCharacter       "%"
TXCharacter       ":"
RXHourCharacter   "r"
TXHourCharacter   "t"

# how units are prefixed when traffic is shown
# 0 = IEC standard prefixes (KiB/MiB/GiB...)
# 1 = old style binary prefixes (KB/MB/GB...)
# 2 = SI decimal prefixes (kB/MB/GB...)
UnitMode 0

# used rate unit (0 = bytes, 1 = bits)
RateUnit 1

# how units are prefixed when traffic rate is shown in bits
# 0 = IEC binary prefixes (Kibit/s...)
# 1 = SI decimal prefixes (kbit/s...)
RateUnitMode 1

# output style
# 0 = minimal & narrow, 1 = bar column visible
# 2 = same as 1 except rate in summary
# 3 = rate column visible
OutputStyle 3
EstimateBarVisible 1

# number of decimals to use in outputs
DefaultDecimals 2
HourlyDecimals 1

# spacer for separating hourly sections (0 = none, 1 = '|', 2 = '][', 3 = '[ ]')
HourlySectionStyle 2

# how many seconds should sampling for -tr take by default
Sampletime 5

# default query mode
# 0 = normal, 1 = days, 2 = months, 3 = top, 5 = short
# 7 = hours, 8 = xml, 9 = one line, 10 = json
QueryMode 0

# default list output entry limits (0 = all)
List5Mins      24
ListHours      24
ListDays       30
ListMonths     12
ListYears       0
ListTop        10


# vnstatd
##

# switch to given user when started as root (leave empty to disable)
DaemonUser ""

# switch to given group when started as root (leave empty to disable)
DaemonGroup ""

# try to detect interface maximum bandwidth, 0 = disable feature
# MaxBandwidth will be used as fallback value when enabled
BandwidthDetection 1

# maximum bandwidth (Mbit) for all interfaces, 0 = disable feature
# (unless interface specific limit is given)
MaxBandwidth 1000

# interface specific limits
#  example 8Mbit limit for eth0 (remove # to activate):
#MaxBWeth0 8

# data retention durations (-1 = unlimited, 0 = feature disabled)
5MinuteHours   48
HourlyDays      4
DailyDays      62
MonthlyMonths  25
YearlyYears    -1
TopDayEntries  20

# how often (in seconds) interface data is updated
UpdateInterval 20

# how often (in seconds) interface status changes are checked
PollInterval 5

# how often (in minutes) data is saved to database
SaveInterval 5

# how often (in minutes) data is saved when all interface are offline
OfflineSaveInterval 30

# on which day should months change
MonthRotate 1
MonthRotateAffectsYears 0

# filesystem disk space check (1 = enabled, 0 = disabled)
CheckDiskSpace 1

# how much the boot time can variate between updates (seconds)
BootVariation 15

# create database entries even when there is no traffic (1 = enabled, 0 = disabled)
TrafficlessEntries 1

# how many minutes to wait during daemon startup for system clock to
# sync time if most recent database update appears to be in the future
TimeSyncWait 5

# how often (in minutes) bandwidth detection is done when
# BandwidthDetection is enabled (0 = disabled)
BandwidthDetectionInterval 5

# force data save when interface status changes (1 = enabled, 0 = disabled)
SaveOnStatusChange 1

# enable / disable logging (0 = disabled, 1 = logfile, 2 = syslog)
UseLogging 2

# create dirs if needed (1 = enabled, 0 = disabled)
CreateDirs 1

# update ownership of files if needed (1 = enabled, 0 = disabled)
UpdateFileOwner 1

# file used for logging if UseLogging is set to 1
LogFile "/var/log/vnstat/vnstat.log"

# file used as daemon pid / lock file
PidFile "/var/run/vnstat/vnstat.pid"

# 1 = 64-bit, 0 = 32-bit, -1 = old style logic, -2 = automatic detection
64bitInterfaceCounters -2

# use SQLite Write-Ahead Logging mode (1 = enabled, 0 = disabled)
DatabaseWriteAheadLogging 0

# change the setting of the SQLite "synchronous" flag
# (-1 = auto, 0 = off, 1, = normal, 2 = full, 3 = extra)
DatabaseSynchronous -1


# vnstati
##

# title timestamp format
HeaderFormat "%Y-%m-%d %H:%M"

# show hours with rate (1 = enabled, 0 = disabled)
HourlyRate 1

# show rate in summary (1 = enabled, 0 = disabled)
SummaryRate 1

# transparent background (1 = enabled, 0 = disabled)
TransparentBg 0

# image size control
LargeFonts 0
LineSpacingAdjustment 0
ImageScale 100

# 5 minutes graph size control
5MinuteGraphResultCount 576
5MinuteGraphHeight 300

# horizontal/vertical summary graph (0 = hours, 1 = 5 minutes)
SummaryGraph 0

# traffic estimate bar style
# (0 = not shown, 1 = continuation of existing bar, 2 = separate bar)
EstimateStyle 1

# bar column in list outputs shows rate if OutputStyle is 3
# (1 = enabled, 0 = disabled)
BarColumnShowsRate 0

# image colors
CBackground     "FFFFFF"
CEdge           "AEAEAE"
CHeader         "606060"
CHeaderTitle    "FFFFFF"
CHeaderDate     "FFFFFF"
CText           "000000"
CLine           "B0B0B0"
CLineL          "-"
CRx             "92CF00"
CTx             "606060"
CRxD            "-"
CTxD            "-"
```

**Comment by vergoh** [comment]
That `vnStat 2.2 config file AUTOMATICALLY GENERATED. DO NOT EDIT!` comment in the /usr/local/etc/vnstat.conf file itself is a little bit strange considering you have version 2.7 in use. The content also hints there's some sort of configuration tool most likely somewhat generating the content and rather likely also discarding possible other inserted configuration options.

How about the actual or even theoreticaly thoughput limit of that PPPoE interface, how fast is it in reality? `vnstat --iflist` should give and idea what it claims to be from the interface side but for correct configuration point of view the real limit from your ISP would also be useful to know.

Based those settings from the output, there doesn't appear to be much deviation from the usual defaults. It's mainly the combination of `BandwidthDetection 1`, `MaxBandwidth 1000` and `UpdateInterval 20` that's likely to be the issue in your case. I suspect that the PPPoE interface doesn't provide any speed information on the interface level and that results in 1000 Mbit being used which likely by far exceeds the actual connection speed. The best way to fix the configuration depends on if you have other interfaces also being monitored, what the output of `vnstat --iflist` shows, what's the actual thoughput limit for that PPPoE interface and if there's some configuration file somewhat which can be modified without it getting overwritten. Depending on the implementation in pfSense, it may also be possible there's an UI somewhere which can be used to input custom settings that then end up in the correct file.

**Comment by Hornochs** [comment]
I guess there is something going on:

```
vnstat --iflist
Available interfaces: igb0 igb1 vtnet0 vtnet1 enc0 pflog0 pfsync0 ovpns1 ovpns2 pppoe0 (64000 Mbit)
```

I can't find anything in the UI. I have all my ifaces monitored and got this issue only on the pppoe0 interface with the statsreset. 

My limit in the pppoe0 interface is 50 Mbit/s down and 10 Mbit/s up. Is there a way to stop the monitoring on vnstat site for a timeperiod?

**Comment by vergoh** [comment]
Add the following lines to /usr/local/etc/vnstat.conf:
```
BandwidthDetection 0
MaxBandwidth 1000
MaxBWpppoe0 50
```
and then remove write permissions from the file and hopefully that way avoid some script trying to modify it at some point: `chmod a-w /usr/local/etc/vnstat.conf`. After that identify the PID of the `vnstatd` process and `kill -HUP` it to force a rescan of the configuration settings or alternatively reboot the system and verify that the content of /usr/local/etc/vnstat.conf still has those changes. That should result in those extra interface resets no longer causing extra traffic from vnStat point of view.

**Comment by kingtat** [comment]
I see something similar with my vnstat installation on pfsense 2.5. Every now and then, a daily upload total of approx 4GB. Actual upload total would be considerably less than this. I also use PPPoE and thought it might have had something to do with interpreation of counters between reboots.
Regarding the config file, there are two packages on pfsense: the freebsd vnstat pkg and the pfsense one pfSense-pkg-Status_Traffic_Totals, providing graphing interface in the UI and also generating the [config file](/pfsense/FreeBSD-ports/blob/641248d8930979e14fd8142ea2e771d12c20f18e/net/pfSense-pkg-Status_Traffic_Totals/files/usr/local/pkg/status_traffic_totals.inc#L33).

**Comment by Hornochs** [comment]
Really interesting to know.

@kingtat did you somehow fixed that for you? I wanted to remove the writeability on the config file in friday, them I'm back Home 😅

**Comment by vergoh** [comment]
@kingtat - For comparison, could you check what sort of output you get with `vnstat --iflist`?
It would be interesting to see if it gives you too 6400 Mbit for the ppoe interface as it did in https://github.com/vergoh/vnstat/issues/224#issuecomment-1039959316

**Comment by kingtat** [comment]
I just wanted to point out where the /usr/local/etc/vnstat.conf file comes from. From glancing at the pfsense package I mentioned, it seems that config file is created if it does not exist but not modified otherwise. The comments in the include file suggest this is to accomodate some future logic for per interface config.
You should be able to edit `/usr/local/etc/vnstat.conf` and have your changes persist. Question remains, what vnstat options to try. I think @vergoh made some suggestions already. I haven't tried them as yet.

**Comment by Hornochs** [comment]
Soooo, I set the changes today and have to wait for the next morning. But directly writing `/usr/local/etc/vnstat.conf` doesn't work! It will be overwritten on every reboot. Which means what @kingtat mentioned in [his comment](https://github.com/vergoh/vnstat/issues/224#issuecomment-1042974829) helped me and I changed the `status_traffic_totals.inc` file. Now there is on a reboot the file written. Which looks to be loaded in the config:

```
# vnStat 2.7 config file
##

# default interface (leave empty for automatic selection)
Interface ""

# location of the database directory
DatabaseDir "/var/db/vnstat"

# locale (LC_ALL) ("-" = use system locale)
Locale "-"

# date output formats for -d, -m, -t and -w
DayFormat    "%Y-%m-%d"
MonthFormat  "%Y-%m"
TopFormat    "%Y-%m-%d"

# characters used for visuals
RXCharacter       "%"
TXCharacter       ":"
RXHourCharacter   "r"
TXHourCharacter   "t"

# how units are prefixed when traffic is shown
# 0 = IEC standard prefixes (KiB/MiB/GiB...)
# 1 = old style binary prefixes (KB/MB/GB...)
# 2 = SI decimal prefixes (kB/MB/GB...)
UnitMode 0

# used rate unit (0 = bytes, 1 = bits)
RateUnit 1

# how units are prefixed when traffic rate is shown in bits
# 0 = IEC binary prefixes (Kibit/s...)
# 1 = SI decimal prefixes (kbit/s...)
RateUnitMode 1

# output style
# 0 = minimal & narrow, 1 = bar column visible
# 2 = same as 1 except rate in summary
# 3 = rate column visible
OutputStyle 3
EstimateBarVisible 1

# number of decimals to use in outputs
DefaultDecimals 2
HourlyDecimals 1

# spacer for separating hourly sections (0 = none, 1 = '|', 2 = '][', 3 = '[ ]')
HourlySectionStyle 2

# how many seconds should sampling for -tr take by default
Sampletime 5

# default query mode
# 0 = normal, 1 = days, 2 = months, 3 = top, 5 = short
# 7 = hours, 8 = xml, 9 = one line, 10 = json
QueryMode 0

# default list output entry limits (0 = all)
List5Mins      24
ListHours      24
ListDays       30
ListMonths     12
ListYears       0
ListTop        10


# vnstatd
##

# switch to given user when started as root (leave empty to disable)
DaemonUser ""

# switch to given group when started as root (leave empty to disable)
DaemonGroup ""

# try to detect interface maximum bandwidth, 0 = disable feature
# MaxBandwidth will be used as fallback value when enabled
BandwidthDetection 0

# maximum bandwidth (Mbit) for all interfaces, 0 = disable feature
# (unless interface specific limit is given)
MaxBandwidth 1000

# interface specific limits
#  example 8Mbit limit for eth0 (remove # to activate):
#MaxBWeth0 8
MaxBWpppoe0 50

# data retention durations (-1 = unlimited, 0 = feature disabled)
5MinuteHours   48
HourlyDays      4
DailyDays      62
MonthlyMonths  25
YearlyYears    -1
TopDayEntries  20

# how often (in seconds) interface data is updated
UpdateInterval 20

# how often (in seconds) interface status changes are checked
PollInterval 5

# how often (in minutes) data is saved to database
SaveInterval 5

# how often (in minutes) data is saved when all interface are offline
OfflineSaveInterval 30

# on which day should months change
MonthRotate 1
MonthRotateAffectsYears 0

# filesystem disk space check (1 = enabled, 0 = disabled)
CheckDiskSpace 1

# how much the boot time can variate between updates (seconds)
BootVariation 15

# create database entries even when there is no traffic (1 = enabled, 0 = disabled)
TrafficlessEntries 1

# how many minutes to wait during daemon startup for system clock to
# sync time if most recent database update appears to be in the future
TimeSyncWait 5

# how often (in minutes) bandwidth detection is done when
# BandwidthDetection is enabled (0 = disabled)
BandwidthDetectionInterval 5

# force data save when interface status changes (1 = enabled, 0 = disabled)
SaveOnStatusChange 1

# enable / disable logging (0 = disabled, 1 = logfile, 2 = syslog)
UseLogging 2

# create dirs if needed (1 = enabled, 0 = disabled)
CreateDirs 1

# update ownership of files if needed (1 = enabled, 0 = disabled)
UpdateFileOwner 1

# file used for logging if UseLogging is set to 1
LogFile "/var/log/vnstat/vnstat.log"

# file used as daemon pid / lock file
PidFile "/var/run/vnstat/vnstat.pid"

# 1 = 64-bit, 0 = 32-bit, -1 = old style logic, -2 = automatic detection
64bitInterfaceCounters -2

# use SQLite Write-Ahead Logging mode (1 = enabled, 0 = disabled)
DatabaseWriteAheadLogging 0

# change the setting of the SQLite "synchronous" flag
# (-1 = auto, 0 = off, 1, = normal, 2 = full, 3 = extra)
DatabaseSynchronous -1


# vnstati
##

# title timestamp format
HeaderFormat "%Y-%m-%d %H:%M"

# show hours with rate (1 = enabled, 0 = disabled)
HourlyRate 1

# show rate in summary (1 = enabled, 0 = disabled)
SummaryRate 1

# transparent background (1 = enabled, 0 = disabled)
TransparentBg 0

# image size control
LargeFonts 0
LineSpacingAdjustment 0
ImageScale 100

# 5 minutes graph size control
5MinuteGraphResultCount 576
5MinuteGraphHeight 300

# horizontal/vertical summary graph (0 = hours, 1 = 5 minutes)
SummaryGraph 0

# traffic estimate bar style
# (0 = not shown, 1 = continuation of existing bar, 2 = separate bar)
EstimateStyle 1

# bar column in list outputs shows rate if OutputStyle is 3
# (1 = enabled, 0 = disabled)
BarColumnShowsRate 0

# image colors
CBackground     "FFFFFF"
CEdge           "AEAEAE"
CHeader         "606060"
CHeaderTitle    "FFFFFF"
CHeaderDate     "FFFFFF"
CText           "000000"
CLine           "B0B0B0"
CLineL          "-"
CRx             "92CF00"
CTx             "606060"
CRxD            "-"
CTxD            "-"
```
Only thing that irrates me:
```
vnstat --iflist
Available interfaces: igb0 igb1 vtnet0 vtnet1 enc0 pflog0 pfsync0 pppoe0 (64000 Mbit) ovpns1 ovpns2
```

Since now pfSense 2.6 is available I can now test, if an upgrade will even destroy the config :D Lets see :) I'll keep you informed with the results.

**Comment by kingtat** [comment]
Yes, I was wrong about that config file :-/
pfSense 2.6 is built on FreeBSD 12.3 which looks to have vnstat-2.8.
There don't appear to have been recent [changes](https://github.com/pfsense/FreeBSD-ports/commits/devel/net/pfSense-pkg-Status_Traffic_Totals) in pfSense-pkg-Status_Traffic_Totals. I'm not sure whether a pfsense upgade reinstalls packages in that case.

**Comment by vergoh** [comment]
> Only thing that irrates me:
> 
> ```
> vnstat --iflist
> Available interfaces: igb0 igb1 vtnet0 vtnet1 enc0 pflog0 pfsync0 pppoe0 (64000 Mbit) ovpns1 ovpns2
> ```

I'm starting to suspect there's a bug somewhere since it's rather unusual to see that many interfaces without interface specific connections speeds know by the BSD kernel. I have two physical FreeBSD 12.2 running machines for testing and neither shows any speeds in the `--iflist` output. However, luckily the raw values are still visible in the debug (`-D`) output so I was able to pulls this out:

System 1 (x86):
```
$ vnstat -D --iflist | grep "getifspeed:"
getifspeed: "msk0": 1000000000
```

System 2 (ppc):
```
$ vnstat -D --iflist | grep "getifspeed:"
getifspeed: "fwe0": 10000000
getifspeed: "gem0": 100000000
```

Based on those values, at least FreeBSD 12 doesn't provide ifi_baudrate in Mbits as the vnStat implementation currently assumes but in bits instead. The vnStat implementation also has a sanity check that if the reported value is larger than 1000000 then the value must be invalid. I'll have to check if previous FreeBSD releases provided ifi_baudrate in Mbits or has this really been broken from the beginning.

What does `vnstat -D --iflist | grep "getifspeed:"` show in pfSense?
I'm not really sure of that 64000 can be right anyway since if it's in bits then that's far too small.

**Comment by Hornochs** [comment]
![grafik](https://user-images.githubusercontent.com/63955136/154903619-a142f972-0209-41f8-a885-addfc7d577ae.png)

Great :D Now it's looking better :) Thanks for the help!

**Comment by vergoh** [comment]
Settings explained just in case someone else ends up here reading this:

- This issue is specific to a BSD based system being used with a PPPoE interface that is being monitored
- `BandwidthDetection 0` - needs to be set to `0` due to a bug in the used vnStat version 2.7 which prevents `MaxBW` settings working if something else is detected for the interface (like is the case in this issue), bug is fixed in vnStat version 2.8 so changing this setting isn't needed with more recent version
- `MaxBandwidth 1000` - sets the default to 1000 Mbit, that's actually the default anyway but doesn't hurt to make it visible here
- `MaxBWpppoe0 50` - sets interface specific limit of 50 Mbit for interface pppoe0 as it is known to be slower than other interfaces in the system

`BandwidthDetection` not working correctly in BSD resulting in the fallback value mostly being used was fixed in commit https://github.com/vergoh/vnstat/commit/c642ea6e08f8eb6707f8ff6d42279c3cb4e6bee2 which will be included in vnStat version 2.10 (or 3.0, whichever it ends up being called).
